azurerm/data_source_arm_scheduled_time_test.go

```diff
@@ -0,0 +1,444 @@
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMScheduledTime_Monthly_Last_Month(t *testing.T) {
+	dataSourceName := "data.azurerm_scheduled_time.test"
+
+	now := time.Now().UTC()
+	scheduletime := now.Add(-25 * time.Duration(24) * time.Hour) //25 days
+	dayofmonth := scheduletime.Day()
+
+	config := testAccDataSourceAzureRMScheduledTime_Monthly(dayofmonth, scheduletime)
+	var expectedTime time.Time
+
+	if now.Day() < 25 {
+		expectedTime = time.Date(scheduletime.Year(), now.Month()+1, scheduletime.Day(), scheduletime.Hour(), 0, 0, 0, time.UTC)
+	} else {
+		expectedTime = time.Date(scheduletime.Year(), now.Month(), scheduletime.Day(), scheduletime.Hour(), 0, 0, 0, time.UTC)
+	}
+	formattedExpectedTime := expectedTime.Format(time.RFC3339)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "day_of_month", strconv.Itoa(dayofmonth)),
+					resource.TestCheckResourceAttr(dataSourceName, "hour", strconv.Itoa(now.Hour())),
+					resource.TestCheckResourceAttr(dataSourceName, "minute", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "second", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "frequency", "Month"),
+					resource.TestCheckResourceAttr(dataSourceName, "next_run_time", formattedExpectedTime),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMScheduledTime_OneTime(t *testing.T) {
+	dataSourceName := "data.azurerm_scheduled_time.test"
+
+	now := time.Now().UTC()
+	scheduletime := now.Add(time.Duration(1) * time.Hour)
+
+	config := testAccDataSourceAzureRMScheduledTime_OneTime(scheduletime)
+	var expectedTime time.Time
+
+	expectedTime = time.Date(scheduletime.Year(), now.Month(), scheduletime.Day(), scheduletime.Hour(), scheduletime.Minute(), 0, 0, time.UTC)
+
+	formattedExpectedTime := expectedTime.Format(time.RFC3339)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "hour", strconv.Itoa(now.Hour()+1)),
+					resource.TestCheckResourceAttr(dataSourceName, "minute", strconv.Itoa(scheduletime.Minute())),
+					resource.TestCheckResourceAttr(dataSourceName, "second", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "frequency", "OneTime"),
+					resource.TestCheckResourceAttr(dataSourceName, "next_run_time", formattedExpectedTime),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAzureRMScheduledTime_Monthly(dayofmonth int, scheduletime time.Time) string {
+	return fmt.Sprintf(`
+data "azurerm_scheduled_time" "test" {
+        "day_of_month" = "%d"
+        "hour" = "%d"
+        "minute" = "0"
+        "second" = "0"
```

Comment:
Yepp, i was thinging about it. I wrote a test but it had problems with asserting. 
If i don't specify the hour, minutes and seconds it will be evaluated from the current time. So the test can't check the `next_run_time` field because it is computed during the tests. I don't have any idea how to avoid this. Is it make sense to write the test without asserting the `next_run_time`? Or do you have any other idea?

It was also a pain to do the assertion for these tests. I am not really satisfied with them. I am wondering to rewrite some of the tests and use conrete times instead of this relaitve ones (eg. now.Add(time.Duration(-1) * time.Minute)) Right now i am not 100% percent sure that these test will always pass, maybe the outcome is time dependent.

Reply:
I fixed some assertion and remove one test which would be fail sometimes (i described the scenario under your comment)

Reply:
Added a test with partially specified schema.